src/components/activities/RosterView.tsx

```diff
@@ -1,102 +1,109 @@
+    // Create a link to download it
+    const anchor = document.createElement('a');
+    anchor.href = url;
+    anchor.setAttribute('download', 'roster.csv');
+    anchor.click();
+    anchor.remove();
   };
 
-  Object.values(activity.participants ?? {}).forEach((participant: Participant) => {
-    buildRosterEntries(participant);
-  });
-
   return (
     <ToolbarPage maxWidth="lg">
-      <Paper>
-        <Table size="small">
-          <TableHead>
-            <TableRow>
-              <TableCell colSpan={7}>
-                <OutputForm columns={2}>
-                  <Stack>
-                    <OutputText label="Name" value={activity?.title}></OutputText>
-                    <OutputText label="State #" value={activity?.idNumber}></OutputText>
-                  </Stack>
-                  <Stack>
-                    <OutputTime time={activity.startTime} label="Start Time"></OutputTime>
-                    <OutputTime time={activity.endTime} label="End Time"></OutputTime>
-                  </Stack>
-                </OutputForm>
-              </TableCell>
-            </TableRow>
-            <TableRow>
-              <TableCell sx={headerCellStyle}>Particpant Name</TableCell>
-              <TableCell sx={headerCellStyle}>Organization</TableCell>
-              <TableCell sx={headerCellStyle}>Sign In</TableCell>
-              <TableCell sx={headerCellStyle}>Arrive Base</TableCell>
-              <TableCell sx={headerCellStyle}>Depart Base</TableCell>
-              <TableCell sx={headerCellStyle}>Sign Out</TableCell>
-              <TableCell sx={headerCellStyle}>Miles</TableCell>
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {rosterEntries.map((entry, i) => (
-              <RosterRow key={i} rosterEntry={entry} activityStartTime={activity.startTime} />
-            ))}
-          </TableBody>
-        </Table>
-      </Paper>
+      <Stack direction="row" flex="1 1 auto" spacing={1} divider={<Divider orientation="vertical" flexItem />}>
+        <Box display="flex" flex="1 1 auto" flexDirection="column">
+          <Paper>{activity ? <Roster activity={activity} rosterEntries={rosterEntries} /> : <ActivityNotFound />}</Paper>
+        </Box>
+        <Stack alignItems="stretch" spacing={2}>
+          <Button variant="outlined" onClick={download}>
+            Download (csv)
+          </Button>
+          <Button variant="outlined" onClick={() => router.push(`/roster/${activityId}/print`)}>
+            Print
+          </Button>
+        </Stack>
+      </Stack>
     </ToolbarPage>
   );
 }
 
+export function RosterPrint({ activityId }: { activityId: string }) {
+  const router = useRouter();
+  const activity = useAppSelector(buildActivitySelector(activityId));
+  const rosterEntries = getRosterEntries(activity);
+  window.onafterprint = () => router.back();
+  setTimeout(() => {
+    window.print();
+  }, 1000);
```

Comment:
Why wait a full second? Without having used the feature, I'm not exactly sure how it feels, but in my mind that's a pretty long time and would surprise a user. Basically they land on the page, wonder what is happening, and then the print dialog magically appears.

Is that just ensuring the page has fully rendered before popping the dialog?

Reply:
Yeah, there might be a better way to do it, but I was having a problem where the print call happens before the component actually renders. I couldn't find a better solution; open to suggestions.

Reply:
[react-to-print](https://www.npmjs.com/package/react-to-print) looks like an option but I was trying to avoid a dependency.